platform/MCU/nRF52840/src/flash_mal.c

```diff
@@ -368,23 +368,31 @@ int FLASH_CopyMemory(flash_device_t sourceDeviceID, uint32_t sourceAddress,
         return FLASH_ACCESS_RESULT_BADARG;
     }
 #if SOFTDEVICE_MBR_UPDATES
-    if (module_function == MODULE_FUNCTION_BOOTLOADER && destinationAddress == USER_FIRMWARE_IMAGE_LOCATION) {
+    if (module_function == MODULE_FUNCTION_BOOTLOADER && destinationAddress == USER_FIRMWARE_IMAGE_LOCATION_COMPAT) {
```

Comment:
Note that `verify_module()` still uses `USER_FIRMWARE_IMAGE_LOCATION` and not `USER_FIRMWARE_IMAGE_LOCATION_COMPAT`.

Reply:
Good catch, thanks!

Reply:
Double checked bootloader updates. Works fine with system-part out of this branch and older releases as well.